extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1199,8 +1197,8 @@ export class AmpA4A extends AMP.BaseElement {
     const iframe = createElementWithAttributes(
```

Comment:
There appears to be a lot of duplicate code between creating xdomain frame via cached or name/safeframe.  Let's modify iframeRenderHelper_ so that it takes an object with the custom attributes to be used.  Something like this:

iframeRenderHelper_(attributes) {
   const mergedAttributes = Object.assign({
       'height': this.element.getAttribute('height'),
       'width': this.element.getAttribute('width'),
   });
   if (this.sentinel) {
     mergedAttributes['data-amp-3p-sentinel'] = this.sentinel;
    }
   const iframe = createElementWithAttributes(
          /** @type {!Document} */(this.element.ownerDocument),
          'iframe', Object.assign(mergedAttributes, SHARED_IFRAME_PROPERTIES));
    protectFunctionWrapper(this.onCreativeRender, this, err => {
      dev().error(TAG, this.element.getAttribute('type'),
          'Error executing onCreativeRender', err);
    })(false);
    // TODO(keithwrightbos): noContentCallback?
    this.xOriginIframeHandler_ = new AMP.AmpAdXOriginIframeHandler(this);
    return this.xOriginIframeHandler_.init(iframe, /* opt_isA4A */ true);
}